exporter/stackdriverexporter/stackdriver.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package stackdriverexporter contains the wrapper for OpenTelemetry-Stackdriver
+// exporter to be used in OpenTelemetry-Service.
+package stackdriverexporter
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"github.com/spf13/viper"
+	"go.opencensus.io/trace"
+
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-service/exporter/exporterwrapper"
+)
+
+type stackdriverConfig struct {
+	ProjectID     string `mapstructure:"project,omitempty"`
+	EnableTracing bool   `mapstructure:"enable_tracing,omitempty"`
+	EnableMetrics bool   `mapstructure:"enable_metrics,omitempty"`
+	MetricPrefix  string `mapstructure:"metric_prefix,omitempty"`
+}
+
+// TODO: Add metrics support to the exporterwrapper.
+type stackdriverExporter struct {
+	exporter *stackdriver.Exporter
+}
+
+var _ consumer.MetricsConsumer = (*stackdriverExporter)(nil)
+
+// StackdriverTraceExportersFromViper unmarshals the viper and returns an consumer.TraceConsumer targeting
```

Comment:
Is this necessary? I already see code with the factory pattern.

Reply:
otoh Start\*Reception and the respective Stop\*Reception methods are needed by the new config.